### ISSUE TITLE
About SSL Certificate Verify

### ISSUE THREAD

**Issue by mYCBS** [sparkle-project/Sparkle#2123]
The server ssl certificate is invalid, is there any way to circumvent sparkle's certificate verification?

**Comment by mYCBS** [comment]
Task <17A40B1B-216A-459C-9196-4F6EA826D3C3>.<1> HTTP load failed, 0/0 bytes (error code: -1202 [3:-9843])
2022-04-27 14:05:52.087903+0800 ASM[20386:289333] Task <17A40B1B-216A-459C-9196-4F6EA826D3C3>.<1> finished with error [-1202] Error Domain=NSURLErrorDomain Code=-1202 "此服务器的证书无效。您可能正在连接一个伪装成“172.29.138.20”的服务器，这会威胁到您的机密信息的安全。" UserInfo={NSLocalizedRecoverySuggestion=您仍想连接该服务器吗？, _kCFStreamErrorDomainKey=3, NSErrorPeerCertificateChainKey=(
    "<cert(0x7fbbc812e400) s: sina.dev i: ZeroSSL RSA Domain Secure Site CA>"
), NSErrorClientCertificateStateKey=0, NSErrorFailingURLKey=https://172.29.138.20:8888/download/appupdate.xml, NSErrorFailingURLStringKey=https://172.29.138.20:8888/download/appupdate.xml, NSUnderlyingError=0x600001312040 {Error Domain=kCFErrorDomainCFNetwork Code=-1202 "(null)" UserInfo={_kCFStreamPropertySSLClientCertificateState=0, kCFStreamPropertySSLPeerTrust=<SecTrustRef: 0x600002f56010>, _kCFNetworkCFStreamSSLErrorOriginalValue=-9843, _kCFStreamErrorDomainKey=3, _kCFStreamErrorCodeKey=-9843, kCFStreamPropertySSLPeerCertificates=(
    "<cert(0x7fbbc812e400) s: sina.dev i: ZeroSSL RSA Domain Secure Site CA>"
)}}, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDownloadTask <17A40B1B-216A-459C-9196-4F6EA826D3C3>.<1>"
), _kCFStreamErrorCodeKey=-9843, _NSURLErrorFailingURLSessionTaskErrorKey=LocalDownloadTask <17A40B1B-216A-459C-9196-4F6EA826D3C3>.<1>, NSURLErrorFailingURLPeerTrustErrorKey=<SecTrustRef: 0x600002f56010>, NSLocalizedDescription=此服务器的证书无效。您可能正在连接一个伪装成“172.29.138.20”的服务器，这会威胁到您的机密信息的安全。}
2022-04-27 14:05:52.088321+0800 ASM[20386:289333] [Sparkle] Encountered download feed error: Error Domain=SUSparkleErrorDomain Code=2001 "下载过程中出现错误，请稍后再试。" UserInfo={NSErrorFailingURLKey=https://172.29.138.20:8888/download/appupdate.xml, NSLocalizedDescription=下载过程中出现错误，请稍后再试。, NSUnderlyingError=0x600001311c20 {Error Domain=NSURLErrorDomain Code=-1202 "此服务器的证书无效。您可能正在连接一个伪装成“172.29.138.20”的服务器，这会威胁到您的机密信息的安全。" UserInfo={NSLocalizedRecoverySuggestion=您仍想连接该服务器吗？, _kCFStreamErrorDomainKey=3, NSErrorPeerCertificateChainKey=(
    "<cert(0x7fbbc812e400) s: sina.dev i: ZeroSSL RSA Domain Secure Site CA>"
), NSErrorClientCertificateStateKey=0, NSErrorFailingURLKey=https://172.29.138.20:8888/download/appupdate.xml, NSErrorFailingURLStringKey=https://172.29.138.20:8888/download/appupdate.xml, NSUnderlyingError=0x600001312040 {Error Domain=kCFErrorDomainCFNetwork Code=-1202 "(null)" UserInfo={_kCFStreamPropertySSLClientCertificateState=0, kCFStreamPropertySSLPeerTrust=<SecTrustRef: 0x600002f56010>, _kCFNetworkCFStreamSSLErrorOriginalValue=-9843, _kCFStreamErrorDomainKey=3, _kCFStreamErrorCodeKey=-9843, kCFStreamPropertySSLPeerCertificates=(
    "<cert(0x7fbbc812e400) s: sina.dev i: ZeroSSL RSA Domain Secure Site CA>"
)}}, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDownloadTask <17A40B1B-216A-459C-9196-4F6EA826D3C3>.<1>"
), _kCFStreamErrorCodeKey=-9843, _NSURLErrorFailingURLSessionTaskErrorKey=LocalDownloadTask <17A40B1B-216A-459C-9196-4F6EA826D3C3>.<1>, NSURLErrorFailingURLPeerTrustErrorKey=<SecTrustRef: 0x600002f56010>, NSLocalizedDescription=此服务器的证书无效。您可能正在连接一个伪装成“172.29.138.20”的服务器，这会威胁到您的机密信息的安全。}}}

**Comment by zorgiepoo** [comment]
This is probably Apple's App Transport Security https://sparkle-project.org/documentation/app-transport-security/

The process to weaken the security this lies in Apple's documentation and some Info.plist keys for your app likely. I'm assuming like most people, you're not using the downloader XPC service (in that case, the keys need to apply to that service bundle).

**Comment by mYCBS** [comment]
![image](https://user-images.githubusercontent.com/11468900/167350506-1689aeba-4319-4f5a-8d63-79040e7ceba6.png)


> 这可能是 Apple 的 App Transport Security https://sparkle-project.org/documentation/app-transport-security/
> 
> 削弱安全性的过程可能在于 Apple 的文档和您的应用程序的一些 Info.plist 密钥。我假设像大多数人一样，您没有使用下载器 XPC 服务（在这种情况下，密钥需要应用于该服务包）。

![image](https://user-images.githubusercontent.com/11468900/167350532-1784dec7-c7c4-4703-aff6-d270f13f85ee.png)
I have set up the project-related settings. There was no problem with http download before, but now such problems will occur when using https. I have consulted the server - the certificate of the server may be invalid. How should I avoid the certificate for such verification now?
